app/src/main/kotlin/com/wire/android/ui/userprofile/service/ServiceDetailsScreen.kt

```diff
@@ -0,0 +1,197 @@
+package com.wire.android.ui.userprofile.service
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material3.Divider
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.wire.android.R
+import com.wire.android.model.ClickBlockParams
+import com.wire.android.ui.common.button.WirePrimaryButton
+import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.snackbar.SwipeDismissSnackbarHost
+import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
+import com.wire.android.ui.home.conversationslist.model.Membership
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.ui.theme.wireDimensions
+import com.wire.android.ui.theme.wireTypography
+import com.wire.android.ui.userprofile.common.EditableState
+import com.wire.android.ui.userprofile.common.UserProfileInfo
+import com.wire.kalium.logic.data.service.ServiceDetails
+import com.wire.kalium.logic.feature.conversation.SecurityClassificationType
+
+@Composable
+fun ServiceDetailsScreen(
+    viewModel: ServiceDetailsViewModel = hiltViewModel()
+) {
+    val snackbarHostState = remember { SnackbarHostState() }
+    val context = LocalContext.current
+
+    LaunchedEffect(Unit) {
+        viewModel.infoMessage.collect {
+            snackbarHostState.showSnackbar(it.asString(context.resources))
+        }
+    }
+
+    ServiceDetailsContent(
+        viewModel = viewModel,
+        snackbarHostState = snackbarHostState
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ServiceDetailsContent(
+    viewModel: ServiceDetailsViewModel,
+    snackbarHostState: SnackbarHostState
+) {
```

Comment:
Can we make `ServiceDetailsContent` not depending on the viewModel by passing the required params ?